src/quicktype-core/language/Dart.ts

```diff
@@ -0,0 +1,353 @@
+function dartNameStyle(startWithUpper: boolean, upperUnderscore: boolean, original: string): string {
+    const words = splitIntoWords(original);
+    return combineWords(
+        words,
+        legalizeName,
+        upperUnderscore ? allUpperWordStyle : startWithUpper ? firstUpperWordStyle : allLowerWordStyle,
+        upperUnderscore ? allUpperWordStyle : firstUpperWordStyle,
+        upperUnderscore || startWithUpper ? allUpperWordStyle : allLowerWordStyle,
+        allUpperWordStyle,
+        "",
+        isStartCharacter
+    );
+}
+
+export class DartRenderer extends ConvenienceRenderer {
+    private _currentFilename: string | undefined;
+    private readonly _gettersAndSettersForPropertyName = new Map<Name, [Name, Name]>();
+
+    constructor(
+        targetLanguage: TargetLanguage,
+        renderContext: RenderContext,
+    ) {
+        super(targetLanguage, renderContext);
+
+    }
+
+    protected forbiddenNamesForGlobalNamespace(): string[] {
+        return keywords;
+    }
+
+    protected forbiddenForObjectProperties(_c: ClassType, _className: Name): ForbiddenWordsInfo {
+        return { names: [], includeGlobalForbidden: true };
+    }
+
+    protected makeNamedTypeNamer(): Namer {
+        return typeNamingFunction;
+    }
+
+    protected namerForObjectProperty(): Namer {
+        return propertyNamingFunction;
+    }
+
+    protected makeUnionMemberNamer(): Namer {
+        return propertyNamingFunction;
+    }
+
+    protected makeEnumCaseNamer(): Namer {
+        return enumCaseNamingFunction;
+    }
+
+    protected unionNeedsName(u: UnionType): boolean {
+        return nullableFromUnion(u) === null;
+    }
+
+    protected namedTypeToNameForTopLevel(type: Type): Type | undefined {
+        // If the top-level type doesn't contain any classes or unions
+        // we have to define a class just for the `FromJson` method, in
+        // emitFromJsonForTopLevel.
+        return directlyReachableSingleNamedType(type);
+    }
+
+    protected makeNamesForPropertyGetterAndSetter(
+        _c: ClassType,
+        _className: Name,
+        _p: ClassProperty,
+        _jsonName: string,
+        name: Name
+    ): [Name, Name] {
+        const getterName = new DependencyName(propertyNamingFunction, name.order, lookup => `get_${lookup(name)}`);
+        const setterName = new DependencyName(propertyNamingFunction, name.order, lookup => `set_${lookup(name)}`);
+        return [getterName, setterName];
+    }
+
+    protected makePropertyDependencyNames(
+        c: ClassType,
+        className: Name,
+        p: ClassProperty,
+        jsonName: string,
+        name: Name
+    ): Name[] {
+        const getterAndSetterNames = this.makeNamesForPropertyGetterAndSetter(c, className, p, jsonName, name);
+        this._gettersAndSettersForPropertyName.set(name, getterAndSetterNames);
+        return getterAndSetterNames;
+    }
+
+    protected startFile(basename: Sourcelike): void {
```

Comment:
At this point I think this should move to `ConvenienceRenderer`.  Feel free to leave it here, I can refactor it later.